external-crates/move/crates/move-binary-format/src/compatibility.rs

```diff
@@ -236,9 +417,10 @@ impl Compatibility {
                 && old_func.visibility != Visibility::Private
                 && old_func.is_entry != new_func.is_entry
             {
-                entry_linking = false
+                context.function_entry_compatibility(old_func, new_func);
             } else if old_func.is_entry && !new_func.is_entry {
-                entry_linking = false;
+                // TODO REVERSE?
```

Comment:
I don't think we need to reverse the order of params here. Initially, I thought that we might want to use different functions for the two cases, but I also think that's not necessary, here's what I think we're checking:

- First case is checking that for pre-bytecode version 5, `public` and `public(package)` functions cannot change whether they are `entry` functions or not.
- Second case is checking that otherwise, a function cannot lose its `entry` status (regardless of its visibility).

So I think we're all good here.